cmake/scripts/osx/ArchSetup.cmake

```diff
@@ -24,6 +24,13 @@ else()
   endif()
 endif()
 
+if(CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "arm64" AND
```

Comment:
Dont use Host processor, thats only relevant to the builder. Use CPU. Its populated in the toolchain file https://github.com/xbmc/xbmc/blob/6e98254fc74ff03876c28a09055bcc022edac119/tools/depends/target/Toolchain.cmake.in#L6